include/irr/asset/filters/CSummedAreaTableImageFilter.h

```diff
@@ -192,40 +178,82 @@ class CSummedAreaTableImageFilter : public CMatchedSizeInOutImageFilterCommon, p
 				std::array<decodeType, maxChannels> maxDecodeValues = {};
 
 				{
-					const auto inData = reinterpret_cast<const uint8_t*>(state->inImage->getBuffer()->getPointer());
+					const uint8_t* inData = reinterpret_cast<const uint8_t*>(state->inImage->getBuffer()->getPointer());
 					const auto blockDims = asset::getBlockDimensions(state->inImage->getCreationParameters().format);
+					static constexpr uint8_t maxPlanes = 4;
+
+					/*
+						Make sure we are able to move as (+ 1) in a certain plane,
+						otherwise memory leaks may occur
+					*/
+
+					bool is2DAndBellow = state->inImage->getCreationParameters().type == IImage::ET_2D;
+					bool is3DAndBellow = state->inImage->getCreationParameters().type == IImage::ET_3D;
+					const core::vectorSIMDu32 limit(1, is2DAndBellow, is3DAndBellow);
+					const core::vectorSIMDu32 movingExclusiveVector = limit, movingOnYZorXZorXYCheckingVector = limit;
 					
 					auto decode = [&](uint32_t readBlockArrayOffset, core::vectorSIMDu32 readBlockPos) -> void
 					{
-						auto newReadBlockPos = (readBlockPos - decltype(readBlockPos)(state->inOffset.x, state->inOffset.y, state->inOffset.z)); // not sure
+						auto newReadBlockPos = (readBlockPos - decltype(readBlockPos)(state->inOffset.x, state->inOffset.y, state->inOffset.z)); 
+						core::vectorSIMDu32 localOutPos = newReadBlockPos * blockDims;
```

Comment:
I'm pretty sure that `state->inOffset` is not counted in blocks.

Also are you sure that `readBlockPos` is counted in blocks?

Reply:
my dear, I'm sure https://github.com/buildaworldnet/IrrlichtBAW/blob/shader_pipeline/include/irr/asset/filters/CBasicImageFilterCommon.h#L39
+ https://github.com/buildaworldnet/IrrlichtBAW/blob/shader_pipeline/include/irr/asset/filters/CBasicImageFilterCommon.h#L32

Reply:
to balance

Reply:
corrected in following commit